core/Pos/Binary/Core/Blockchain.hs

```diff
@@ -59,3 +64,26 @@ instance ( Typeable b
         body   <- decode
         extra  <- decode
         toCborError $ T.recreateGenericBlock header body extra
+
+----------------------------------------------------------------------------
+-- BlockHeader
+----------------------------------------------------------------------------
+
+instance ( HasConfiguration
+         , T.BlockchainHelpers MainBlockchain
+         , T.BlockchainHelpers GenesisBlockchain
+         ) =>
+         Bi BlockHeader where
+   encode x = encodeListLen 2 <> encodeWord tag <> body
+     where
+       (tag, body) = case x of
+         BlockHeaderGenesis bh -> (0, encode bh)
+         BlockHeaderMain bh    -> (1, encode bh)
+
+   decode = do
+       decodeListLenCanonicalOf 2
+       t <- decodeWordCanonical
+       case t of
+           0 -> BlockHeaderGenesis <$!> decode
```

Comment:
Why you used strict fmap here?

Reply:
Just preserving the semantics, the code was strict before my changes.